[WEBSITE-375] Document withCredentials (along with adding a fleshing out credential-handling content in general)

gilesgas opened the pull request:
For details, see https://issues.jenkins-ci.org/browse/WEBSITE-375.

Includes a new section **Handling credentials** in the [Using a Jenkinsfile](https://jenkins.io/doc/book/pipeline/jenkinsfile/) page and a new page **Using credentials** in the [Using Jenkins](https://jenkins.io/doc/book/using/) chapter.

This PR will also finalize some User Handbook restructuring that was discussed in https://issues.jenkins-ci.org/browse/WEBSITE-400.

bitwiseman commented:
@rtyler @abayer  Should it be "Declarative Pipeline" or "declarative Pipeline"?

bitwiseman commented on content/doc/book/pipeline/using-credentials.adoc:
```diff
@@ -0,0 +1,134 @@
+---
```
Move this to using Jenkins

bitwiseman commented on content/doc/book/pipeline/jenkinsfile.adoc:
```diff
@@ -25,18 +27,18 @@ provides a number of immediate benefits:
 
-Pipeline supports <<syntax#, two syntaxes>>, Declarative (introduced in
-Pipeline 2.5) and Scripted Pipeline. Both of which support building continuous
+Pipeline supports <<syntax#, two syntaxes>>, declarative (introduced in
```
If we want to change the case of these it should be a separate PR and done everywhere.

gilesgas replied:
Now reverted

bitwiseman commented on content/doc/book/pipeline/jenkinsfile.adoc:
```diff
@@ -403,18 +411,383 @@ apply to all steps within the Pipeline.
+keys or certificates, then use Jenkins' *Snippet Generator* feature, which you
+can access through Jenkins' classic UI.
+
+To access the *Snippet Generator* for your Pipeline project/item:
```
Do we need a general Snippet Generator page/section?  If you had that, this section could be a little less detailed.   Probably need a separate PR/task to cover that change. 

As it stands, the is good, but would need to change if/when the Snippet Generate pages is created.

gilesgas replied:
Yes that's a good point ... when I next include a procedure that makes use of the Snippet Generator on this 'Using a Jenkinsfile' page (or another other relevant page in the 'Pipeline' chapter), then I'll extract this information into its own section near the top of this 'Using a Jenkinsfile' page.

gilesgas commented:
Hello @bitwiseman , I've addressed your comments above so this is now be ready to merge.

I've refactored and removed some text in the introductory sections to each chapter so that these initial paragraphs better explain the content of their respective chapters.

I also changed the title of the first page "Getting Started with Jenkins" to "User Handbook Overview" and updated the content to reflect the new title, since the original title no longer seemed to ring true.

gilesgas commented:
Okay - this is really done now.

bitwiseman commented on content/doc/book/pipeline/jenkinsfile.adoc:
```diff
@@ -230,10 +234,11 @@ node {
 `sh` step always sees a zero exit code, giving the `junit` step the opportunity
 to capture and process the test reports. Alternative approaches to this are
-covered in more detail in the <<handling-failures>> section below.
+covered in more detail in the <<failure-handling>> section below.
```
`failure-handling` an anchor label that I can see.

gilesgas replied:
Whoops! Thanks @bitwiseman - this was a remnant of my playing around with gerunds and imperatives for heading titles and I forgot to revert this back when I decided that gerunds were the way forward for heading titles in the User Handbook.
